### DISCUSSION TITLE
Diversity reception with RSPduo

### DISCUSSION THREAD

**Discussion by flocentblack** [AlexandreRouma/SDRPlusPlus#595] [Ideas]
Hi Alexandre,

I have been watching this project with interest since its first release.  I bought an RSPduo solely for the purpose of 2 channel reception of the same frequency for stereo/binaural listening. Unfortunately SDRuno does not run on linux, nor does it have this feature.

While considering buying an Afedri  2 channel SDR, I came across some forum posts that were unfair and unkind to you, which completely turned me off to buying an Afredri SDR, especially since I like what you are doing.  My only option is to find a way to use the RSP duo.

The ability to combine and make phase and amplitude adjustment would be nice, but my primary need is to be able to listen to tuner 1 in my left ear and tuner 2 in my right, with the tuners locked together at the same frequency, with co-located  horizontal and vertically polarized antennas, to minimize fading in HF ham radio reception.

While this is probably not a feature that has a huge demand, I am seeing more and more interest in this type of reception, and a number of other SDR software platforms offer this. With the exception of linrad, they are all closed source and do not run on linux, and there is no software that will do this with an RSPduo.

I'm sure you get a lot of feature requests, If this is something that you are not interested in pursuing, I understand. Let me know if I can be of any help to you in this, or if there is any guidance you could offer for someone who has limited coding skills in modifying SDR++ for this feature.

Thanks for taking the time to read this,

Eric / NQ8J

**Comment by AlexandreRouma** [comment]
Hello,
Unfortunately, due to the design of the signal path in SDR++ this is impossible to implement. It wasn't designed to have several IQ paths and making it possible would require basically a complete rewrite and redesign of the entire app.

Regards.

**Reply by flocentblack** [reply]
Ok thanks.

73 de NQ8J

**Reply by rskunath** [reply]
You can do what you are asking already with SDR Uno. Just open Uno in Dual Mode (Master/Slave). In this mode you have 2 instances of SDR Uno and you can have one antenna on one instance and the other on the second instance and control audio distribution so you have one tuner in each ear separately. You can get the 2 instances to track using Omnirig.

**Reply by potty-dzmeia** [reply]
Maybe this is something worth considering. You can do some really cool stuff using diversity reception. 
Actually, I am also looking for the same think - I call it stereo diversity reception.

**Comment by flocentblack** [comment]
Thanks. Unforunately, being that two instances of SDR Uno are not phase coherent,  there can be a signifacant latency between two instances of SDR Uno, which depending on the computer, can be enough of delay to be an audible echo.  

I’ve got this working in GNU radio, but the GUI is nowhere nearly as nice as SDR++ :-)

**Reply by potty-dzmeia** [reply]
Here is the description of how to do this in SDRuno: https://www.youtube.com/watch?v=drhOZPY68bI

However, as flocentblack mentioned the tuners are not phase coherent and what you hear is not "true" but somewhat random stereo diversity reception.

**Comment by rskunath** [comment]
Are you using an SDR Duo?

**Comment by flocentblack** [comment]
Yea.

Eric

**Comment by rskunath** [comment]
I just ran an experiment here. Set the Duo to Dual (normal) mode In SDR Uno and started another instance of SDR Uno. That came up in Slave mode. I then selected a mediumwave channel, set the VRX's on both Master and Slave to the same mode and frequency and I routed Master audio to the left ear and Slave audio to the right ear. It worked fine and I couldn't hear any sort of echo.

I tuned the pair around a lot and always they seemed to work as expected. I especially tried to listen for echoes etc., but I couldn't hear any.

I'd be curious to hear your thoughts?

**Reply by potty-dzmeia** [reply]
This is what the creators are saying about the mode you mention ([link ](https://www.sdrplay.com/community/viewtopic.php?f=7&t=3254&start=20)to thread):

_There seems to be a bit of confusion here. The two tuners in the RSPduo both run off a common reference clock and so if tuned to the same frequency, both tuners will be phase locked and I/Q streams are time aligned. However, because as things currently stand, the two tuners (to operate simultaneously) have to operate in separate instances of SDRuno, then as the instances cannot be started simultaneously, buffering means that the audio for the two tuners will not be time aligned between the separate instances of SDRuno._

So this is not an actual phase coherent mode.
